### DISCUSSION TITLE
Understanding how ectd interacts with its WAL

### DISCUSSION THREAD

**Discussion by sebastiaoamaro** [etcd-io/etcd#17702] [Q&A] [answered]
Hi everyone,
I am studying how etcd nodes interact with the WAL file, when it writes to it, which system calls are called etc.

I am running the https://github.com/etcd-io/etcd/blob/main/Procfile example and using strace to trace each individual node. However, I do not see any write system calls to the WAL file in fact looking at the strace logs I see this:

```
openat(AT_FDCWD, "/tmp/infra2/", O_RDONLY|O_CLOEXEC) = 30
fcntl(30, F_GETFL)                      = 0x8000 (flags O_RDONLY|O_LARGEFILE)
fcntl(30, F_SETFL, O_RDONLY|O_NONBLOCK|O_LARGEFILE) = 0
epoll_ctl(4, EPOLL_CTL_ADD, 30, {events=EPOLLIN|EPOLLOUT|EPOLLRDHUP|EPOLLET, data={u32=2000683052, u64=9167902785335722028}}) = -1 EPERM (Operation not permitted)
fcntl(30, F_GETFL)                      = 0x8800 (flags O_RDONLY|O_NONBLOCK|O_LARGEFILE)
fcntl(30, F_SETFL, O_RDONLY|O_LARGEFILE) = 0
getdents64(30, 0xc00229e000 /* 4 entries */, 8192) = 144
getdents64(30, 0xc00229e000 /* 0 entries */, 8192) = 0
close(30)   
```
/tmp/infra2 is the wall directory I passed, I see an operation not permitted error. However, I am able to still run simple put and simple transactions in the cluster. 

I tried with both the 3.3 version and the most recent one, Is there something wrong with my configuration am I misunderstanding how the WAL works?

Thanks in advance.

**Comment by jmhbnz** [comment, marked as answer]
Hey @sebastiaoamaro - Thanks for your question.  To try and determine if there are any configuration issues, can you please check the logs on each member for any errors?

As an aside, this docs page is a good resource for learning more of how etcd uses the WAL files https://etcd.io/docs/v3.5/learning/persistent-storage-files.

**Reply by sebastiaoamaro** [reply]
Sorry for the late reply! I did not find that when I was searching, will take a look!
